### ISSUE TITLE
Training and Inference discussion for baseline model

### ISSUE THREAD

**Issue by rohanbanerjee** [sct-pipeline/fmri-segmentation#34]
#### What is the baseline model
The model which was trained on ✅ as per the QCs mentioned in #25 is the `baseline` model. A total of 96 images were used in the training of this model. A list of subjects (for later reference) is below:
[participants_baseline.csv](https://github.com/sct-pipeline/fmri-segmentation/files/14561835/participants_baseline.csv)

The model was trained in 6 different settings - 5 models for the 5 fold cross-validation and 1 fold_all model (discussion can be found here - https://github.com/MIC-DKFZ/nnUNet/issues/1364#issuecomment-1492075312). The config (containing preprocessing, hyperparameters) for nnUNetv2 training is: [plans.json](https://github.com/sct-pipeline/fmri-segmentation/files/14561924/plans.json)

After the model was trained, the inference was run on the ❌  (failed segmentations) from #25. Below are the QCs from all the folds and fold_all:

[qc_fold_0.zip](https://github.com/sct-pipeline/fmri-segmentation/files/14562125/qc_fold_0.zip)
[qc_fold_1.zip](https://github.com/sct-pipeline/fmri-segmentation/files/14562234/qc_fold_1.zip)
[qc_fold_2.zip](https://github.com/sct-pipeline/fmri-segmentation/files/14562357/qc_fold_2.zip)
[qc_fold_3.zip](https://github.com/sct-pipeline/fmri-segmentation/files/14562363/qc_fold_3.zip)
[qc_fold_4.zip](https://github.com/sct-pipeline/fmri-segmentation/files/14562374/qc_fold_4.zip)
[qc_fold_all.zip](https://github.com/sct-pipeline/fmri-segmentation/files/14562413/qc_fold_all.zip)

The steps to reproduce the above QC results (/run inference) are the following:
1. Clone this repos
2. `cd fmri-segmentation`
3. Download the model weights (the whole folder) from the link: https://drive.google.com/drive/folders/1WSn-15wGWz6i2_aZeQTwKls2sZ6dpfHf?usp=share_link
4. Install dependencies
```
pip install -r run_nnunet_inference_requirements.txt
```
5. Run the command:
```
python run_nnunet_inference.py --path-dataset <PATH TO FOLDER CONTAINING IMAGES, SUFFIXED WITH _0000> --path-out <PATH TO O/P FOLDER> --path-model <PATH TO DOWNLOADED WEIGHTS FOLDER>
``` 

Next steps:
- [x] Choose the fold whose inference results will be used for the next round of training
- [x] Choose and correct ~30 segmentations and include it in the next training round along with the ones that have good segmentation after inference.
- [ ]  Get metric by running evaluation on the `held-out test set` (#33)

**Comment by rohanbanerjee** [comment]
I chose 30 subjects from the `fold_all` inference that I manually corrected. The list of 30 chosen subjects is below:
[qc_report_fold_all.zip](https://github.com/sct-pipeline/fmri-segmentation/files/14603262/qc_report_fold_all.zip)

I manually corrected the segmentation for these subjects and attaching the QC for the same below:
[qc_fold_all_corrected.zip](https://github.com/sct-pipeline/fmri-segmentation/files/14605057/qc_fold_all_corrected.zip)


After @jcohenadad approves these segmentations, I'll add these images to the training set and start the re-training.

**Comment by jcohenadad** [comment]
review: [qc_fail.yml.zip](https://github.com/sct-pipeline/fmri-segmentation/files/14608054/qc_fail.yml.zip)

details: 

spinefmri-sub-genevaR104_003_0000.nii.gz
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/837568bd-b161-4d22-9d30-262cfc7b7ec8)

spinefmri-sub-genevaR106_005_0000.nii.gz
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/0dc86d02-61c9-4ec5-9be9-0888e45935f8)

spinefmri-sub-genevaR108_007_0000.nii.gz
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/3cd97bc2-f18f-489e-9706-1f33648b6068)

spinefmri-sub-genevaR202_018_0000.nii.gz
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/79661fb6-cc0d-4ce9-96ca-26de05b21628)
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/9f6d3182-36a3-495c-b6f7-2cca568a600a)

spinefmri-sub-genevaR207_023_0000.nii.gz
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/26953156-7ba4-4419-b3c8-40968527c06f)

spinefmri-sub-genevaR209_025_0000.nii.gz
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/c81b3c24-3c0e-4210-b02a-35886e53d83b)

spinefmri-sub-genevaR211_027_0000.nii.gz
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/41a9d8e2-58f7-435b-bf46-43805d8f4303)

spinefmri_sub-nwM09_task-motor_bold_0000.nii.gz
overseg: ![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/c4cb1096-0caa-4063-bf10-f6d4388444ed)

spinefmri_sub-nwM16_task-motor_bold_0000.nii.gz
![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/7fd88164-0e85-4abb-991d-fcb0cef7542d)

"Download all" button is not there-- please make sure to use the latest version of SCT:

![image](https://github.com/sct-pipeline/fmri-segmentation/assets/2482071/e3d1fa1c-ad90-4b8d-b161-0bc3d3c2d074)

**Comment by rohanbanerjee** [comment]
Closing the issue since the baseline model training was successfully completed (including running inference and manual correction)
